ops/testing.py

```diff
@@ -149,6 +163,39 @@ def _create_meta(self, charm_metadata, action_metadata):
 
         return charm.CharmMeta.from_yaml(charm_metadata, action_metadata)
 
+    def add_oci_resource(self, resource_name: str, contents: typing.Mapping = None) -> None:
```

Comment:
is it a generic Mapping, or do the keys and values need to be strings? If the latter, than maybe a `Mapping[str, str]` is more appropriate.

Reply:
Yup I believe it's always str, str I'll update this.